GitWorkflow/GitWorkflow/Player.cpp

```diff
@@ -21,19 +21,28 @@ void Player::collide(const Platform& platform)
 
 void Player::update(const float dt)
 {
-	m_velocity.x = 0.0f;
+	if (std::roundf(m_velocity.y) == 0.0f && (sf::Keyboard::isKeyPressed(sf::Keyboard::LShift) || sf::Keyboard::isKeyPressed(sf::Keyboard::RShift))) m_speed = m_baseSpeed * 1.5f;
+	else m_speed = m_baseSpeed;
 
-	if (sf::Keyboard::isKeyPressed(sf::Keyboard::A)) m_velocity.x -= m_speed;
-	if (sf::Keyboard::isKeyPressed(sf::Keyboard::D)) m_velocity.x += m_speed;
+	if (sf::Keyboard::isKeyPressed(sf::Keyboard::A)) { if (std::abs(m_velocity.x) < m_speed * (std::roundf(m_velocity.y) == 0.0f ? 1.0f : 0.5f)) m_velocity.x -= m_acceleration * dt * (std::roundf(m_velocity.y) == 0.0f ? 1.0f : 0.5f); }
+	else if (m_velocity.x < 0.0f) m_velocity.x += m_acceleration * dt;
 
-	if (sf::Keyboard::isKeyPressed(sf::Keyboard::Space) && !m_hasJumped)
+	if (sf::Keyboard::isKeyPressed(sf::Keyboard::D)) { if (std::abs(m_velocity.x) < m_speed * (std::roundf(m_velocity.y) == 0.0f ? 1.0f : 0.5f)) m_velocity.x += m_acceleration * dt * (std::roundf(m_velocity.y) == 0.0f ? 1.0f : 0.5f); }
+	else if (m_velocity.x > 0.0f) m_velocity.x -= m_acceleration * dt;
+
+	if (sf::Keyboard::isKeyPressed(sf::Keyboard::Space))
 	{
-		m_velocity.y -= m_speed;
-		m_hasJumped = true;
+		if (!m_hasJumped)
+		{
+			m_velocity.y = -m_baseSpeed * 1.25f;
+			m_hasJumped = true;
+		}
 	}
 
-	move(sf::Vector2f(0.0f, 9.82f) * (dt * dt / 2.0f));
-	m_velocity.y += 9.82f * dt;
+	else if (m_hasJumped && m_velocity.y < 0.0f) { m_velocity.y += m_speed * dt;  }
```

Comment:
This looks strange. Only applying this velocity while the player is moving up?

Reply:
It should only apply when the player is ascending ergo when his vertical velocity is negative. Otherwise it will lead to the player falling faster, unless the Space button is being held down.